app/src/main/java/com/T05/krowdtrialz/ui/experimentDetails/ExperimentDetailsActivity.java

```diff
@@ -105,8 +108,13 @@ public void onFailure() {
                 Log.e(TAG, "Error Searching Database for experiment");
             }
         });
+    }
 
-
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        // Stop listening to changes in the Database
+        expRegistration.remove();
```

Comment:
Note: It's good to call this after getExperimentByID, once we're done with the snapshot listener